Makefile

```diff
@@ -1,69 +1,37 @@
 SHELL = /bin/bash
 
-##
-# Definitions.
-.SUFFIXES:
+# Environments
+export VCS_URL          := $(shell git config --get remote.origin.url)
+export VCS_REF          := $(shell git rev-parse --short HEAD)
+export BUILD_DATE       := $(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
 
 
+## Targets ----
 
-## Tools.
-tools =
-
-ifeq ($(shell uname -s),Darwin)
-	SED = gsed
-else
-	SED = sed
-endif
-
-DOCKER = docker
-DOCKER_COMPOSE = docker-compose
-
-
-## Targets .
-.PHONY: .env
-.env:
-	@echo VCS_REF=$(shell git rev-parse --short HEAD)           >.env
-	@echo VCS_URL=$(shell git config --get remote.origin.url)  >>.env
-	@echo BUILD_DATE=$(shell date -u +"%Y-%m-%dT%H:%M:%SZ")    >>.env
+.PHONY: help
+help: ## This help.
+	@sort $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-.]+:.*?## / {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' 
 
-.PHONY: all
-# target: all – Builds all images
-all: build
+.DEFAULT_GOAL := help
 
 .PHONY: build
-# target: build – Builds all images (uses cache)
-build: .env 
-	${DOCKER_COMPOSE} build
-
-.PHONY: rebuild
-# target: rebuild – Builds all images from scratch
-rebuild: .env 
-	${DOCKER_COMPOSE} build --no-cache
+build:  ## Builds all images (uses cache)
+	docker-compose build
 
+.PHONY: build-nc
+build-nc: ## Builds all images from scratch
+	docker-compose build --no-cache
 
 .PHONY: deploy
-# target: deploy – Tags images and pushes to dockerhub's registry [TODO]
-deploy: .env 
+publish:  ## TODO: Tags images and pushes to dockerhub's registry
 	echo TODO: tagging images and pushing to dockerhub ...
-	#${DOCKER_COMPOSE} push
+	#docker-compose push
```

Comment:
Of late I started to add a safeguard to make sure one could only tag an image if on origin/master ... 

```
uptodate-test: ## Have all the changes been commited ?
        @ git fetch upstream
        @if [ "$$(git diff upstream/master)" != "" ]; then \
                echo "*** Can't tag/release/publish if you are not in sync with upstream/master ***"; \
                exit 1; \
        fi
```

and then protext the tag target in this way

```
tag-latest: uptodate-test ## Tag easii-prod:latest container
        @echo 'create tag latest'
        docker tag $(APP_NAME):$(VERSION) $(DOCKER_REPO)/$(APP_NAME):latest
```

Reply:
will consider in #16